src/main/java/nl/tudelft/contextproject/model/level/roomIO/TileParser.java

```diff
@@ -43,13 +60,37 @@ public static void readTiles(MazeTile[][] tiles, int width, int height, int xOff
 			for (int x = 0; x < line.length; x++) {
 				int posx = x + xOffset;
 				int posy = y + yOffset;
-				if (line[x].equals("#")) {
+				if (line[x].equals(EMPTY_FORMAT)) {
 					// always overwrite old value
 					tiles[posx][posy] = null;
 				} else {
-					tiles[posx][posy] = new MazeTile(posx, posy, TileType.valueOf(line[x]));
+					tiles[posx][posy] = new MazeTile(posx, posy, format(line[x]));
 				}
 			}
 		}
 	}
+
+	/**
+	 * Translate the map file to actual type.
+	 * @param string
+	 * 			read type
+	 * @return
+	 * 			TileType value
+	 */
+	public static TileType format(String string) {
+		switch (string) {
+			case WALL_FORMAT :
+				return TileType.WALL;
+			case FLOOR_FORMAT:
+				return TileType.FLOOR;
+			case DOOR_ENTRANCE_FORMAT:
+				return TileType.DOOR_ENTRANCE;
+			case DOOR_EXIT_FORMAT:
+				return TileType.DOOR_EXIT;
+			case CORRIDOR_FORMAT:
+				return TileType.CORRIDOR;
+			default:
+				throw new IllegalArgumentException("Incorrect maze tile in maze file: " + string);
+		}
```

Comment:
The TileParser defines it's own format to translate the room map format. Thus it wouldn't make sense to move this method to an enum, as different kind of parsers can use different translator methods.